src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -328,11 +295,12 @@ public static void Reverse<T>(this Span<T> span)
         public static int IndexOf<T>(this ReadOnlySpan<T> span, T value)
             where T : IEquatable<T>
         {
-            if (typeof(T) == typeof(byte))
+            if (IsTypeNumeric<T>(out int size))
                 return SpanHelpers.IndexOf(
                     ref Unsafe.As<T, byte>(ref MemoryMarshal.GetReference(span)),
                     Unsafe.As<T, byte>(ref value),
-                    span.Length);
+                    ((nuint)span.Length) * size);
```

Comment:
If I am reading this correctly, this is going to return byte index now. Doesn't it need to return the actual item index?

A lot of tests should be failing because of this.

Reply:
Yes, and they are. Fixing it now.